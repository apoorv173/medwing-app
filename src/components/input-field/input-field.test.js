import React from 'react';
import { shallow } from 'enzyme';
import InputField from './input-field';
import renderer from 'react-test-renderer';

it('Input onchange event', () => {
	// Render a checkbox with label in the document
	const onChangeFn = jest.fn();
	
	const inputField = shallow(<InputField 
			name="search-input"
			onChange={onChangeFn}
			ariaDescribedBy="Enter location"
			placeholder= "Enter location"
			label="Enter location"
		/>
	);

	inputField.find('input').simulate('change', 'berlin');
	expect(onChangeFn.mock.calls.length).toEqual(1);
  expect(onChangeFn).toBeCalledWith('berlin');
});

it('renders correctly', () => {
  const tree = renderer
    .create(<InputField 
			name="search-input"
			ariaDescribedBy="Enter location"
			placeholder= "Enter location"
			label="Enter location"
		/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});