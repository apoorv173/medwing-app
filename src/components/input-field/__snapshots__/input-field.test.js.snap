// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="search-input"
  >
    Enter location
  </label>
  <input
    className="form-control"
    name="search-input"
    onChange={[Function]}
    placeholder="Enter location"
    type="text"
  />
</div>
`;
